devlab/tests/functional_test.py

```diff
@@ -149,8 +149,18 @@ def filter_vms(self):
         return [i for i in self.src_cloud.novaclient.servers.list(**opts)
                 if i.name in vms_names]
 
-    def filter_flavors(self):
-        flavors = [i['name'] for i in config.flavors]
+    def filter_flavors(self, filter_only_private=False):
+        flavors = []
+        all_flavors = config.flavors
+        for tenant in config.tenants:
+            if tenant.get('flavors'):
+                all_flavors += [flavor for flavor in tenant['flavors']]
+        for flavor in all_flavors:
+            if filter_only_private:
+                if 'is_public' in flavor and not flavor.get('is_public'):
+                    flavors.append(flavor['name'])
+            elif 'is_public' not in flavor or flavor.get('is_public'):
+                flavor.append(flavor['name'])
         return self._get_nova_resources('flavors', flavors)
 
     def filter_keypairs(self):
```

Comment:
To get all flavors (public and non-public) need to pass additional key is_public=None:
self.novaclient.flavors.list(is_public=None).